src/app/dependency-injection/using-injector/base-class-for-injector/base-class-for-injector.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Injector, OnInit } from '@angular/core';
+import { LoggerService } from '../../../../shared/logger.service';
+
+@Component({
+  selector: 'app-base-class-for-injector',
+  templateUrl: './base-class-for-injector.component.html',
+})
+export class BaseClassForInjectorComponent implements OnInit {
+  readonly logger: LoggerService;
```

Comment:
You can also add `protected` in the beginning.
Shortly, what it means: when you have here `public` (default value), you can access this service in inherited component templates, or by ViewChild from outer parent components of these inherited components which breaks the logic of encapsulation a bit.
But when you set `protected`, you will have only and only access inside the inherited component TS file, and it will be not available inside HTML file, and any other places too. So, makes your logic protected.